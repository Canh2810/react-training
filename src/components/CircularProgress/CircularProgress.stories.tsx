import { StoryObj, Meta } from '@storybook/react'

import CircularProgress from './'

export default {
  title: 'Components/CircularProgress',
  component: CircularProgress,
} as Meta

type Story = StoryObj<typeof CircularProgress>

export const Default: Story = {}
