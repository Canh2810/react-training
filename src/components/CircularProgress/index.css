.loader {
  border: 7px solid var(--border-color);
  border-radius: 50%;
  border-top: 7px solid var(--primary-color);
  width: 40px;
  height: 40px;
  -webkit-animation: spin 2s linear infinite; /* Safari */
  animation: spin 2s linear infinite;
}

/* Safari */
@-webkit-keyframes spin {
  0% {
    -webkit-transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
