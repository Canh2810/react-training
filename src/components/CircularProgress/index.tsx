import './index.css'

const CircularProgress = () => <div className="loader"></div>

export default CircularProgress
