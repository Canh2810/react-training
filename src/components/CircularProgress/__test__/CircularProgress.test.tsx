import { render } from '@testing-library/react'
import CircularProgress from '..'

describe('CircularProgress component', () => {
  it('should render correctly', () => {
    const { container } = render(<CircularProgress />)
    expect(container).toMatchSnapshot()
  })
})
