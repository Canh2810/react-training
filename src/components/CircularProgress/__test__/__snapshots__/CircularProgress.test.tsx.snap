// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress component should render correctly 1`] = `
<div>
  <div
    class="loader"
  />
</div>
`;
